### PR TITLE
P4-2260 improving test coverage primarily for the importers including some other tidy ups.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
These changes are primarily adding tests were they were missing e.g the importers.  Also includes some minor tidy ups in others as well.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/MovePrice.kt [hunk 1]
```diff
@@ -1,7 +1,5 @@
 package uk.gov.justice.digital.hmpps.pecs.jpc.calculator
 
-import uk.gov.justice.digital.hmpps.pecs.jpc.location.Location
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/MovePrice.kt:3]: Just removing unused imports.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsSpreadsheet.kt [hunk 2]
```diff
@@ -21,7 +21,7 @@ class LocationsSpreadsheet(private val spreadsheet: Workbook, private val locati
     init {
         val missingTabs = Tab.values().filter { spreadsheet.getSheet(it.label) == null }.toList()
 
-        if (missingTabs.isNotEmpty()) throw NullPointerException("The following tabs are missing from the locations spreadsheet: ${missingTabs.joinToString { it.label }}")
+        if (missingTabs.isNotEmpty()) throw RuntimeException("The following tabs are missing from the locations spreadsheet: ${missingTabs.joinToString { it.label }}")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsSpreadsheet.kt:24]: Changed all exceptions to be runtimes for simplicity.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheet.kt [hunk 3]
```diff
@@ -37,20 +37,22 @@ class PricesSpreadsheet(private val spreadsheet: Workbook,
...
+        val journeyId = Result.runCatching { row.getCell(JOURNEY_ID).numericCellValue }.getOrElse { throw RuntimeException("Error retrieving journey id for supplier '$supplier'", it) }
         val fromLocationName = row.getFormattedStringCell(FROM_LOCATION)
-                ?: throw NullPointerException("From location name cannot be blank")
+                ?: throw RuntimeException("From location name cannot be blank")
         val toLocationName = row.getFormattedStringCell(TO_LOCATION)
-                ?: throw NullPointerException("To location name cannot be blank")
-        val price = Result.runCatching { row.getCell(PRICE).numericCellValue }.getOrElse { throw IllegalArgumentException("Error retrieving price for supplier '$supplier'", it) }
+                ?: throw RuntimeException("To location name cannot be blank")
+        val price = Result.runCatching { (row.getCell(PRICE).numericCellValue * 100).toInt() }
+                .onSuccess { if (it == 0) throw RuntimeException("Price must be greater than zero") }
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheet.kt:46]: Including additional check to make sure price is greater than zero.  This is enforced on the JPA entity.  This makes it a little easier to consume if the error occurs.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt [hunk 4]
```diff
@@ -59,28 +60,44 @@ internal class PricesSpreadsheetTest {
...
             row.getCell(1).setCellValue("")
 
             assertThatThrownBy { spreadsheet.mapToPrice(row) }
-                    .isInstanceOf(NullPointerException::class.java)
+                    .isInstanceOf(RuntimeException::class.java)
                     .hasMessage("From location name cannot be blank")
         }
 
+        @Test
+        internal fun `throws error if price is zero`() {
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt:86]: New test to check price is greater than zero,

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt [hunk 5]
```diff
@@ -59,28 +60,44 @@ internal class PricesSpreadsheetTest {
...
-            whenever(locationRepository.findBySiteName(any())).thenReturn(location)
-            whenever(priceRepository.findByFromLocationNameAndToLocationName(any(), any())).thenReturn(null)
+            whenever(locationRepository.findBySiteName("FROM SITE")).thenReturn(fromLocation)
+            whenever(locationRepository.findBySiteName("TO SITE")).thenReturn(toLocation)
+            whenever(priceRepository.findByFromLocationNameAndToLocationName("FROM SITE", "TO SITE")).thenReturn(null)
 
-            assertThatCode { spreadsheet.mapToPrice(row) }.doesNotThrowAnyException()
+            val price = spreadsheet.mapToPrice(row)
+
+            assertThat(price.fromLocationId).isEqualTo(fromLocation.id)
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheetTest.kt:69]: Expanded to test actual values as before we could not say for certain.

**Review by jcavell — APPROVED** [review]

**Pull request merged**